snap/system_usernames,tests: Azure IoT Edge system usernames

alexclewontin opened the pull request:
Genericize system-usernames-microk8s test to test arbitrary usernames scoped to a specific snap ID. Then add Azure IoT Edge scoped system usernames.

alexclewontin commented:
Checks failed because I messed up the PR name, should be good now

mardy commented:
Thanks Alex! I like the changes on the spread test!

I'm not sure about adding all those usernames for a single snap, it seems a bit too much, but if you already agreed this with @pedronis then it's fine for me as well. In any case, I don't think that we need to add unit tests for all of them; I actually wouldn't add any unit test, given that the functionality is already tested by the existing usernames, and you have a good test in the spread.

alexclewontin commented:
@mardy thanks for the review! 

I'll nix the unit tests. 

Just as context: I also don't love adding all the usernames but unfortunately it is necessary to support the Azure IoT Edge application architecture, which relies on all of these services running under different UIDs corresponding to known usernames (which we are meeting Microsoft halfway with by adding the `snap_` prefix).

alexclewontin commented:
Split the usernames across two snap IDs (one is an Azure ID service which can be used by more than just the Azure IoT Edge runtime, the other is the Azure IoT Edge runtime). 

Removed unnecessary unit tests as per @mardy's review.

Looks like the failing check is not related to this PR?

pedronis reviewed (dismissed):
this looks reasonable, including the parameterized spread tests, thanks. If the use cases grow further we need to make this policy controlled from the store though

pedronis commented on snap/system_usernames.go:
```diff
@@ -155,4 +155,19 @@ var SupportedSystemUsernames = map[string]systemUsername{
 	"snap_microk8s": {Id: 584789, AllowedSnapIds: []string{
 		"EaXqgt1lyCaxKaQCU349mlodBkDCXRcg", // microk8s
 	}},
+	"snap_iotedge": {Id: 584790, AllowedSnapIds: []string{
```
why is this not snap_aziotedge ?

alexclewontin replied:
Because the azure-iot-edge codebase uses a different naming convention than the azure-iot-identity codebase (the one that uses the aziot prefixes). This is intentional and `snap_iotedge` is the correct name here but if this is absolutely not workable I can talk to the Microsoft folks

pedronis reviewed (commented):
actually a question about the first username

alexclewontin commented:
> this looks reasonable, including the parameterized spread tests, thanks. If the use cases grow further we need to make this policy controlled from the store though

@pedronis unfortunately I already know of one more request that is going to come through the pipe (the Azure Device Update service, a different Microsoft service which also uses the azure-iot-identity snap as an authentication backend).

If that would be too many we can look at reverting to store-side declaration method

alexclewontin commented:
Added the usernames for the above-mentioned Azure Device Update service. Check failure appears unrelated.

mardy reviewed (approved):
I'll let Samuele answer your question, but code-wise this LGTM, thanks!

mvo5 reviewed (approved):
Thanks!

pedronis reviewed (approved):
thanks
